src/kem/kyber768/field_element.rs

```diff
@@ -0,0 +1,111 @@
+use std::ops;
+
+use crate::kem::kyber768::{parameters::FIELD_MODULUS, utils::field::FieldElement};
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct KyberFieldElement {
+    pub value: u16,
+}
+
+impl KyberFieldElement {
+    pub const MODULUS: u16 = FIELD_MODULUS;
+
+    const BARRETT_SHIFT : u32 = 24; // 2 * ceil(log_2(FIELD_MODULUS))
+    const BARRETT_MULTIPLIER : u32 = (1u32 << Self::BARRETT_SHIFT) / (Self::MODULUS as u32);
+
+    pub fn barrett_reduce(value : u32) -> Self {
+        let product : u64 = u64::from(value) * u64::from(Self::BARRETT_MULTIPLIER);
+        let quotient : u32 = (product >> Self::BARRETT_SHIFT).try_into().unwrap();
```

Comment:
Using `as` everywhere here gives you a 25% performance improvement.
`unwrap` is pretty expensive.